storage/Devices/PartitionImpl.cc

```diff
@@ -133,11 +133,29 @@ namespace storage
     unsigned int
     Partition::Impl::get_number() const
     {
-	string::size_type pos = get_name().find_last_not_of("0123456789");
-	if (pos == string::npos || pos == get_name().size() - 1)
+	const string& name = get_name();
+
+	string::size_type pos = name.find_last_not_of("0123456789");
+	if (pos == string::npos || pos == name.size() - 1)
 	    ST_THROW(Exception("partition name has no number"));
 
-	return atoi(get_name().substr(pos + 1).c_str());
+	return atoi(name.substr(pos + 1).c_str());
+    }
+
+
+    void
+    Partition::Impl::set_number(unsigned int number)
+    {
+	const string& name = get_name();
+
+	string::size_type pos = name.find_last_not_of("0123456789");
```

Comment:
For symmetry reasons, there should also be the other way round: A format() function to build a partition device name from its base device name ("/dev/sda") and its number. Yes, it sounds trivial, but it's a matter of centralizing tests and having good unit test coverage.